pkg/frontend/v2/frontend.go

```diff
@@ -127,10 +127,11 @@ type queryResultWithBody struct {
 }
 
 type frontendRequest struct {
-	queryID      uint64
-	request      *httpgrpc.HTTPRequest
-	userID       string
-	statsEnabled bool
+	queryID         uint64
+	httpRequest     *httpgrpc.HTTPRequest
+	protobufRequest proto.Message
```

Comment:
Note to reviewers: this isn't actually set anywhere yet, so all the existing code paths should behave as they did before, and always use a HTTP request payload.